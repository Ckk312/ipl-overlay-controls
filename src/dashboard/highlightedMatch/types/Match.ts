export interface Match{
    meta: {
        name: string;
        id: string;
        stageName?: string;
        completeTime?: string;
        round?: number;
        match?: number;
    };
    teamA: {
        id?: string;
        name?: string;
        logoUrl?: string;
        players?: {
            name?: string;
        }[];
    };
    teamB: {
        id?: string;
        name?: string;
        logoUrl?: string;
        players?: {
            name?: string;
        }[];
    };
}