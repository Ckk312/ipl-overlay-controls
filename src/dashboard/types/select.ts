export type SelectOptions = Array<{ name: string, value: string }>;
