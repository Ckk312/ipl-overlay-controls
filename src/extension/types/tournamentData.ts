/* tslint:disable */
/**
 * This file was generated using http://json2ts.com/ with sample Battlefy Data.
 */

export interface Rules {
    complete: string;
    critical: string;
}

export interface CustomField {
    name: string;
    public: boolean;
    _id: string;
}

export interface Contact {
    type: string;
    details: string;
}

export interface EmailsSent {
    oneDay: boolean;
    now: boolean;
}

export interface BattlefyTournamentData {
    _id: string;
    startTime: Date;
    rules: Rules;
    playersPerTeam: number;
    customFields: CustomField[];
    userCanReport: boolean;
    name: string;
    about: string;
    bannerUrl: string;
    contact: Contact;
    contactDetails: string;
    prizes: string;
    schedule: string;
    type: string;
    checkInRequired: boolean;
    checkInStarts: string;
    hasRegistrationCap: boolean;
    teamCap: number;
    countryFlagsOnBrackets: boolean;
    isPublished: boolean;
    hasMaxPlayers: boolean;
    serviceFeePercent: number;
    gameName: string;
    isFeatured: boolean;
    isPublic: boolean;
    isSuspended: boolean;
    isRosterLocked: boolean;
    registrationEnabled: boolean;
    createdAt: Date;
    updatedAt: Date;
    checkInStartTime: Date;
    slug: string;
    cloudSearchDocumentHash: string;
    cloudSearchDocumentLastGenerated: Date;
    clonedFromTournamentID: string;
    allowedCountries: boolean;
    maxPlayers: number;
    emailsSent: EmailsSent;
    organizationID: string;
    gameID: string;
    stageIDs: string[];
    streamIDs: string[];
    hasPassword: boolean;
    lastCompletedMatchAt: Date;
    templateID: string;
}