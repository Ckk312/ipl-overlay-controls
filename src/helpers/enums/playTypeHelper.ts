import { PlayType } from '../../types/enums/playType';

export class PlayTypeHelper {
    static toPrettyString(value: PlayType, gameCount?: number): string {
        const count = !gameCount ? '' : ` ${gameCount}`;
        return {
            [PlayType.PLAY_ALL]: `Play all${count}`,
            [PlayType.BEST_OF]: `Best of${count}`
        }[value];
    }

    static fromBattlefySeriesStyle(value: string): PlayType | undefined {
        switch (value?.toLowerCase()) {
            case 'gamespermatch':
                return PlayType.PLAY_ALL;
            case 'bestof':
                return PlayType.BEST_OF;
            default:
                return undefined;
        }
    }
}
