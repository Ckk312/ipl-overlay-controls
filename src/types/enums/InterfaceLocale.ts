export enum InterfaceLocale {
    EN = 'EN',
    JA = 'JA'
}
