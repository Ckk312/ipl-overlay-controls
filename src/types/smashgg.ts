export interface SmashggEvent {
    id: number
    name: string
    game: string
}

export interface SmashggEventsResponse {
    data: {
        tournament: {
            id: number
            events: {
                id: number
                name: string
                videogame: {
                    displayName: string
                }
            }[]
        }
    }
}

export interface SmashggEntrantsResponse {
    data: {
        event: {
            id: number
            name: string
            videogame: {
                displayName: string
            }
            tournament: {
                name: string
                id: number
                slug: string
            }
            entrants: {
                pageInfo: {
                    total: number
                    totalPages: number
                }
                nodes: {
                    id: number
                    name: string
                    team: {
                        images: {
                            url: string
                            type: 'profile'
                        }[]
                    }
                    participants: {
                        id: number
                        prefix?: string
                        gamerTag: string
                        user?: {
                            genderPronoun?: string
                        }
                    }[]
                }[]
            }
        }
    }
}
