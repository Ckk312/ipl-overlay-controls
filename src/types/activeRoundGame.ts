import { GameWinner } from 'types/gameWinner';

export interface ActiveRoundGame {
    winner: GameWinner;
    stage: string;
    mode: string;
    color?: {
        index: number;
        title: string;
        clrA: string;
        clrB: string;
        categoryName: string;
        colorsSwapped: boolean;
    };
}
