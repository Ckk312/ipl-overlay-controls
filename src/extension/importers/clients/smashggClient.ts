import { Team } from 'types/team';
import axios from 'axios';
import { HighlightedMatches, TournamentData } from 'schemas';
import { TournamentDataSource } from 'types/enums/tournamentDataSource';
import {
    SmashggEntrantsResponse,
    SmashggEvent,
    SmashggEventsResponse,
    SmashggTournamentStreamQueueResponse,
    SmashggTournamentStreamQueueSlot
} from 'types/smashgg';
import isEmpty from 'lodash/isEmpty';

export async function getSmashGGEvents(slug: string, token: string): Promise<SmashggEvent[]> {
    const query = `query Events($slug: String!) {
      tournament(slug: $slug) {
        id
        events {
          id
          name
          videogame {
            displayName
          }
        }
      }
    }`;

    const response = await axios.post<SmashggEventsResponse>(
        'https://api.smash.gg/gql/alpha',
        JSON.stringify({
            query,
            variables: {
                slug
            }
        }),
        {
            headers: {
                'Content-Type': 'application/json',
                Accept: 'application/json',
                Authorization: `Bearer ${token}`
            }
        }
    );

    return response.data.data.tournament.events.map(event => ({
        id: event.id,
        name: event.name,
        game: event.videogame.displayName
    }));
}

export async function getSmashGGData(eventId: number, token: string): Promise<TournamentData> {
    const firstPageData = await getSmashGGPage(1, eventId, token, true);
    const slug = firstPageData.raw.data.event.tournament.slug;

    const tourneyInfo: TournamentData = {
        meta: {
            id: slug.substring(slug.indexOf('/') + 1),
            source: TournamentDataSource.SMASHGG,
            name: firstPageData.raw.data.event.tournament.name,
            sourceSpecificData: {
                smashgg: {
                    tournamentId: firstPageData.raw.data.event.tournament.id,
                    streams: firstPageData.raw.data.event.tournament.streams,
                    eventData: {
                        id: firstPageData.raw.data.event.id,
                        name: firstPageData.raw.data.event.name,
                        game: firstPageData.raw.data.event.videogame.displayName
                    }
                }
            }
        },
        teams: firstPageData.pageInfo
    };

    // If there are more pages, add them to our data set
    if (firstPageData.raw.data.event.entrants.pageInfo.totalPages > 1) {
        const pagePromises = [];
        for (let i = 2; i <= firstPageData.raw.data.event.entrants.pageInfo.totalPages; i++) {
            pagePromises.push(getSmashGGPage(i, eventId, token));
        }

        const pages = await Promise.all(pagePromises);

        for (let i = 0; i < pages.length; i++) {
            tourneyInfo.teams = tourneyInfo.teams.concat(pages[i].pageInfo);
        }
    }

    return tourneyInfo;
}

async function getSmashGGPage(
    page: number,
    eventId: number,
    token: string,
    getRaw = false
): Promise<{ pageInfo: Team[], raw?: SmashggEntrantsResponse }> {
    const query = `query EventEntrants($eventId: ID!, $page: Int!, $perPage: Int!) {
      event(id: $eventId) {
        id
        videogame {
          displayName
        }
        tournament {
          id
          name
          slug
          streams{
            id
            streamName
          }
        }
        name
        entrants(query: {
          page: $page
          perPage: $perPage
        }) {
          pageInfo {
            total
            totalPages
          }
          nodes {
            id
            name
            team {
              images {
                type
                url
              }
            }
            participants {
              id
              prefix
              gamerTag
              user{
                genderPronoun
              }
            }
          }
        }
      }
    }`;
    const perPage = '50';

    const response = await axios.post<SmashggEntrantsResponse>(
        'https://api.smash.gg/gql/alpha',
        JSON.stringify({
            query,
            variables: {
                eventId,
                page,
                perPage
            }
        }),
        {
            headers: {
                'Content-Type': 'application/json',
                Accept: 'application/json',
                Authorization: `Bearer ${token}`
            }
        }
    );

    const { data } = response;
    const pageInfo: Team[] = [];
    data.data.event.entrants.nodes.forEach(entrant => {
        pageInfo.push({
            id: entrant.id.toString(10),
            name: entrant.name,
            showLogo: true,
            logoUrl: entrant.team?.images.find(image => image.type === 'profile')?.url,
            players: entrant.participants.map(participant => ({
                name: isEmpty(participant.prefix)
                    ? participant.gamerTag : `${participant.prefix} ${participant.gamerTag}`,
                pronouns: participant.user?.genderPronoun?.toLowerCase()
            }))
        });
    });

    if (getRaw) {
        return { pageInfo, raw: data };
    } else {
        return { pageInfo };
    }
}

/**
 * Creates a Team object from SmashGG slot data
 * @param teamData Slot data for a team
 */
function streamQueueTeamCreator(teamData: SmashggTournamentStreamQueueSlot): Team {
    const team: Team = {
        id: teamData.entrant.id.toString(),
        name: teamData.entrant.name,
        showLogo: true,
        seed: teamData.seed.groupSeedNum,
        players: teamData.entrant.participants.map(participant => ({
            name: isEmpty(participant.prefix)
                ? participant.gamerTag : `${participant.prefix} ${participant.gamerTag}`,
            pronouns: participant.user?.genderPronoun?.toLowerCase()
        }))
    };
    // Check if team object is present
    if (teamData.entrant.team !== null) {
        team.logoUrl = teamData.entrant.team.images.find(image => image.type === 'profile')?.url;
    }
    return team;
}

/**
 * Get Queued sets
 * @param slug Tournament slug
 * @param token Smash.gg token
 * @param streamIDs List of streamID you want to get sets for. If blank, all all queued sets will be returned
 */
export async function getSmashGGStreamQueue(
    slug: string,
    token: string,
    streamIDs: number[]
): Promise<HighlightedMatches> {
    const query = `query EventEntrants($slug: String!) {
        tournament(slug: $slug) {
            streamQueue {
                stream{
                  id
                }
                sets {
                    id
                    round
                    event {
                        name
                        videogame{
                          slug
                        }
                    }
                    phaseGroup {
                        displayIdentifier
                    }
                    slots {
                        seed {
                            groupSeedNum
                        }
                        entrant {
                            id
                            name
                            team {
                                images {
                                    url
                                    type
                                }
                            }
                            participants {
                                prefix
                                gamerTag
                                user {
                                    genderPronoun
                                }
                            }
                        }
                    }
                }
            }
        }
    }`;

    const response = await axios.post<SmashggTournamentStreamQueueResponse>(
        'https://api.smash.gg/gql/alpha',
        JSON.stringify({
            query,
            variables: {
                slug,
            }
        }),
        {
            headers: {
                'Content-Type': 'application/json',
                Accept: 'application/json',
                Authorization: `Bearer ${token}`
            }
        }
    );

    const highlightedStreamQueue: HighlightedMatches = [];
    const { data } = response;

    data.data.tournament.streamQueue.forEach(streamQueue => {
        // If steamIDs is not empty && the streamIDs include this streamID for the queue
        // or is streamIDs is empty
        if ((streamIDs.length !== 0 && streamIDs.includes(streamQueue.stream.id)) || streamIDs.length === 0) {
            streamQueue.sets.forEach(set => {
                // If there's 2 teams and neither of the objects are null
                if (set.slots.length === 2 && !(set.slots.some(slot => slot.entrant === null))) {
                    highlightedStreamQueue.push({
                        meta: {
                            id: set.id.toString(),
                            stageName: set.phaseGroup.displayIdentifier,
                            round: set.round,
                            match: 0,
                            name: `${set.phaseGroup.displayIdentifier} Round ${set.round}`
                        },
                        teamA: streamQueueTeamCreator(set.slots[0]),
                        teamB: streamQueueTeamCreator(set.slots[1])
                    });
                }
            });
        }
    });
    return highlightedStreamQueue;
}

