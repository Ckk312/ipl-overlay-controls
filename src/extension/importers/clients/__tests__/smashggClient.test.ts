import axios from 'axios';
import { getSmashGGData, getSmashGGEvents } from '../smashggClient';
import { TournamentDataSource } from 'types/enums/tournamentDataSource';
import { SmashggEntrantsResponse } from 'types/smashgg';

describe('smashggClient', () => {
    jest.mock('axios');
    const mockPost = jest.fn();

    beforeEach(() => {
        jest.resetAllMocks();

        axios.post = mockPost;
    });

    describe('getSmashGGEvents', () => {
        it('fetches events from Smash.gg', async () => {
            mockPost.mockResolvedValue({
                data: {
                    data: {
                        tournament: {
                            events: [
                                {
                                    id: 123431423,
                                    name: 'Cool Event',
                                    videogame: {
                                        displayName: 'Splatoon 2'
                                    }
                                },
                                {
                                    id: 238470278,
                                    name: 'Cool Smash',
                                    videogame: {
                                        displayName: 'Ultimate'
                                    }
                                }
                            ]
                        }
                    }
                }
            });

            const result = await getSmashGGEvents('event', 'apdjwiadhjwio');

            expect(result).toEqual([
                {
                    id: 123431423,
                    name: 'Cool Event',
                    game: 'Splatoon 2'
                },
                {
                    id: 238470278,
                    name: 'Cool Smash',
                    game: 'Ultimate'
                }
            ]);
            expect(mockPost).toHaveBeenCalledWith(
                'https://api.smash.gg/gql/alpha',
                expect.any(String),
                {
                    headers: {
                        'Content-Type': 'application/json',
                        Accept: 'application/json',
                        Authorization: 'Bearer apdjwiadhjwio'
                    }
                }
            );
            expect(cleanUpGraphqlQuery(mockPost.mock.calls[0][1])).toMatchSnapshot();
        });
    });

    describe('getSmashGGData', () => {
        it('fetches data from Smash.gg and turns it into our expected format', async () => {
            mockPost.mockResolvedValue({
                data: {
                    data: {
                        event: {
                            id: 1234567,
                            name: 'Splatoon Two',
                            videogame: {
                                displayName: 'Splatoon 2'
                            },
                            tournament: {
                                name: 'Cool Tournament',
                                id: 654423,
                                slug: 'ct'
                            },
                            entrants: {
                                pageInfo: {
                                    total: 10,
                                    totalPages: 1
                                },
                                nodes: [
                                    {
                                        id: 1111111,
                                        name: 'Team One',
                                        team: {
                                            images: [
                                                {
                                                    url: 'smashgg://img',
                                                    type: 'profile'
                                                }
                                            ]
                                        },
                                        participants: [
                                            {
                                                id: 12345,
                                                prefix: 'TO',
                                                gamerTag: 'Player One',
                                                user: {
                                                    genderPronoun: 'he/him'
                                                }
                                            },
                                            {
                                                id: 23456,
                                                gamerTag: 'Player Two',
                                                user: { }
                                            }
                                        ]
                                    },
                                    {
                                        id: 2222222,
                                        name: 'Team Two',
                                        participants: [
                                            {
                                                id: 45678,
                                                gamerTag: 'Player Three',
                                                user: { }
                                            },
                                            {
                                                id: 23456,
                                                prefix: 'TT',
                                                gamerTag: 'Player Four',
                                                user: {
                                                    genderPronoun: 'She/Her'
                                                }
                                            }
                                        ]
                                    }
                                ]
                            }
                        }
                    }
                } as SmashggEntrantsResponse
            });

            const result = await getSmashGGData(123123, '149083257830574');

            expect(result).toEqual({
                meta: {
                    id: 'ct',
                    name: 'Cool Tournament',
                    source: TournamentDataSource.SMASHGG,
                    sourceSpecificData: {
                        smashgg: {
                            eventData: {
                                game: 'Splatoon 2',
                                id: 1234567,
                                name: 'Splatoon Two'
                            },
                            tournamentId: 654423
                        }
                    }
                },
                teams: [
                    {
                        id: '1111111',
                        name: 'Team One',
                        logoUrl: 'smashgg://img',
                        showLogo: true,
                        players: [
                            { name: 'TO Player One', pronouns: 'he/him' },
                            { name: 'Player Two', pronouns: undefined }
                        ]
                    },
                    {
                        id: '2222222',
                        name: 'Team Two',
                        showLogo: true,
                        logoUrl: undefined,
                        players: [
                            { name: 'Player Three', pronouns: undefined },
                            { name: 'TT Player Four', pronouns: 'she/her' }
                        ]
                    }
                ]
            });
            expect(mockPost).toHaveBeenCalledWith(
                'https://api.smash.gg/gql/alpha',
                expect.any(String),
                {
                    headers: {
                        'Content-Type': 'application/json',
                        Accept: 'application/json',
                        Authorization: 'Bearer 149083257830574'
                    }
                }
            );
            expect(cleanUpGraphqlQuery(mockPost.mock.calls[0][1])).toMatchSnapshot();
        });

        it('fetches data multiple times if more than one page is available', async () => {
            mockPost.mockResolvedValueOnce({
                data: {
                    data: {
                        event: {
                            id: 1234567,
                            name: 'Splatoon Two',
                            videogame: {
                                displayName: 'Splatoon 2'
                            },
                            tournament: {
                                name: 'Cool Tournament',
                                id: 654423,
                                slug: 'ct'
                            },
                            entrants: {
                                pageInfo: {
                                    total: 10,
                                    totalPages: 3
                                },
                                nodes: [
                                    {
                                        id: 1111111,
                                        name: 'Team One',
                                        team: {
                                            images: [
                                                {
                                                    url: 'smashgg://img',
                                                    type: 'profile'
                                                }
                                            ]
                                        },
                                        participants: [
                                            {
                                                id: 12345,
                                                prefix: 'TO',
                                                gamerTag: 'Player One',
                                                user: {
                                                    genderPronoun: 'he/him'
                                                }
                                            },
                                            {
                                                id: 23456,
                                                gamerTag: 'Player Two',
                                                user: { }
                                            }
                                        ]
                                    }
                                ]
                            }
                        }
                    }
                } as SmashggEntrantsResponse
            }).mockResolvedValueOnce({
                data: {
                    data: {
                        event: {
                            id: 1234567,
                            name: 'Splatoon Two',
                            videogame: {
                                displayName: 'Splatoon 2'
                            },
                            tournament: {
                                name: 'Cool Tournament',
                                id: 654423,
                                slug: 'ct'
                            },
                            entrants: {
                                pageInfo: {
                                    total: 10,
                                    totalPages: 3
                                },
                                nodes: [
                                    {
                                        id: 345345345,
                                        name: 'Cool Team',
                                        team: {
                                            images: [
                                                {
                                                    url: 'smashgg://img_cool',
                                                    type: 'profile'
                                                }
                                            ]
                                        },
                                        participants: [
                                            {
                                                id: 12345,
                                                prefix: 'CT',
                                                gamerTag: 'Player One',
                                                user: {
                                                    genderPronoun: 'he/him'
                                                }
                                            },
                                            {
                                                id: 23456,
                                                gamerTag: 'Player Two',
                                                user: { }
                                            }
                                        ]
                                    },
                                    {
                                        id: 56756757,
                                        name: 'Team Three',
                                        team: {
                                            images: []
                                        },
                                        participants: [
                                            {
                                                id: 45678,
                                                gamerTag: 'Player Three',
                                                user: { }
                                            },
                                            {
                                                id: 23456,
                                                prefix: 'TT',
                                                gamerTag: 'Player Four',
                                                user: {
                                                    genderPronoun: 'She/Her'
                                                }
                                            }
                                        ]
                                    }
                                ]
                            }
                        }
                    }
                } as SmashggEntrantsResponse
            }).mockResolvedValueOnce({
                data: {
                    data: {
                        event: {
                            id: 1234567,
                            name: 'Splatoon Two',
                            videogame: {
                                displayName: 'Splatoon 2'
                            },
                            tournament: {
                                name: 'Cool Tournament',
                                id: 654423,
                                slug: 'ct'
                            },
                            entrants: {
                                pageInfo: {
                                    total: 10,
                                    totalPages: 3
                                },
                                nodes: [
                                    {
                                        id: 89089,
                                        name: 'Team Four',
                                        team: {
                                            images: []
                                        },
                                        participants: [
                                            {
                                                id: 45678,
                                                gamerTag: 'Player Three',
                                                user: { }
                                            },
                                            {
                                                id: 23456,
                                                prefix: 'TT',
                                                gamerTag: 'Player Four',
                                                user: {
                                                    genderPronoun: 'She/Her'
                                                }
                                            }
                                        ]
                                    }
                                ]
                            }
                        }
                    }
                } as SmashggEntrantsResponse
            });

            const result = await getSmashGGData(1234, '149083257830574');

            expect(result).toEqual({
                meta: {
                    id: 'ct',
                    name: 'Cool Tournament',
                    source: TournamentDataSource.SMASHGG,
                    sourceSpecificData: {
                        smashgg: {
                            eventData: {
                                game: 'Splatoon 2',
                                id: 1234567,
                                name: 'Splatoon Two'
                            },
                            tournamentId: 654423
                        }
                    }
                },
                teams: [
                    {
                        id: '1111111',
                        name: 'Team One',
                        showLogo: true,
                        logoUrl: 'smashgg://img',
                        players: [
                            { name: 'TO Player One', pronouns: 'he/him' },
                            { name: 'Player Two', pronouns: undefined }
                        ]
                    },
                    {
                        id: '345345345',
                        name: 'Cool Team',
                        logoUrl: 'smashgg://img_cool',
                        showLogo: true,
                        players: [
                            { name: 'CT Player One', pronouns: 'he/him' },
                            { name: 'Player Two', pronouns: undefined }
                        ]
                    },
                    {
                        id: '56756757',
                        name: 'Team Three',
                        logoUrl: undefined,
                        showLogo: true,
                        players: [
                            { name: 'Player Three', pronouns: undefined },
                            { name: 'TT Player Four', pronouns: 'she/her' }
                        ]
                    },
                    {
                        id: '89089',
                        name: 'Team Four',
                        logoUrl: undefined,
                        showLogo: true,
                        players: [
                            { name: 'Player Three', pronouns: undefined },
                            { name: 'TT Player Four', pronouns: 'she/her' }
                        ]
                    }
                ]
            });
            expect(mockPost).toHaveBeenCalledTimes(3);
        });
    });
});

function cleanUpGraphqlQuery(query: string): string {
    return query
        .replace(/\\n/g, '')
        .replace(/ {4,}/g, ' ');
}
