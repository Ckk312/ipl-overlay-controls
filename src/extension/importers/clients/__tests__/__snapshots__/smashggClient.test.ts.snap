// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`smashggClient getSmashGGData fetches data from Smash.gg and turns it into our expected format 1`] = `"{\\"query\\":\\"query EventEntrants($eventId: ID!, $page: Int!, $perPage: Int!) { event(id: $eventId) { id videogame { displayName } tournament { id name slug } name entrants(query: { page: $page perPage: $perPage }) { pageInfo { total totalPages } nodes { id name team { images { type url } } participants { id prefix gamerTag user{ genderPronoun } } } } } }\\",\\"variables\\":{\\"eventId\\":123123,\\"page\\":1,\\"perPage\\":\\"50\\"}}"`;

exports[`smashggClient getSmashGGEvents fetches events from Smash.gg 1`] = `"{\\"query\\":\\"query Events($slug: String!) { tournament(slug: $slug) { id events { id name videogame { displayName } } } }\\",\\"variables\\":{\\"slug\\":\\"event\\"}}"`;
